### PR TITLE
Added image text extraction (OCR) extensions

### PR DESCRIPTION
**Pull request by pavelgj**

### CODE REVIEW

**Comment by elvisun** [comment]
Just an idea, what if we can write to RTDB or Firestore as another option?

**Review by jhuleatt — CHANGES_REQUESTED** [review]
Thanks @pavelgj! Can you please add docs in the form of `PREINSTALL.md` and `POSTINSTALL.md` files? Here are some samples: [preinstall](https://raw.githubusercontent.com/firebase/firebase-tools/master/templates/extensions/PREINSTALL.md), [postinstall](https://raw.githubusercontent.com/firebase/firebase-tools/master/templates/extensions/POSTINSTALL.md), [dynamic links extension](https://github.com/FirebaseExtended/experimental-extensions/tree/next/firestore-shorten-urls-dynamic-links)

**Comment by jhuleatt** [comment]
> Just an idea, what if we can write to RTDB or Firestore as another option?

Agreed, same for #8. Some thoughts on the options:

1. Support text file in Storage only (keep it as-is)
    - **Pros**: No additional permissions required, doesn't assume developer uses another Firebase product
    - **Cons**: Developer has to write their own code to download text file and parse it, slower than reading from DB
1. Support RTDB only, or Firestore only
    - **Pros**: Simpler implementation, 
    - **Cons**: Only works with a single DB, doesn't work for developers that don't use that DB, need to add additional permissions to Extension to allow it to write to a DB
1. Support RTDB, Firestore, and writing to a text file in Storage (settable via param)
    - **Pros**: Works great for developers that use RTDB and Firestore, and is fine even for those that don't use a Firebase DB
    - **Cons**: More complex implementation, extension would require permission for multiple products (even if a developer doesn't use all of them)

I lean toward the second two options, but the first one works too! It's up to you @pavelgj.

**Comment by mbleigh** [comment]
My vote would be for (2) with Firestore as the only option (for now). If developers are installing an extension to OCR a file, chances are they want to have the content of that file more readily accessible than re-downloading the file would make it.

Separate note: we should really figure out how to have "variants" of an extension (e.g. a Firestore and RTDB version of this one).

**Comment by pavelgj** [comment]
> Thanks @pavelgj! Can you please add docs in the form of `PREINSTALL.md` and `POSTINSTALL.md` files? Here are some samples: [preinstall](https://raw.githubusercontent.com/firebase/firebase-tools/master/templates/extensions/PREINSTALL.md), [postinstall](https://raw.githubusercontent.com/firebase/firebase-tools/master/templates/extensions/POSTINSTALL.md), [dynamic links extension](https://github.com/FirebaseExtended/experimental-extensions/tree/next/firestore-shorten-urls-dynamic-links)

Done.

**Comment by pavelgj** [comment]
Updated the extension to write extracted text to Firestore.

File: storage-extract-image-text/PREINSTALL.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+This extension will extract text from any `jpg` or `png` images uploaded to Firebae Storage and write the extracted text to Firestore.
```
Comment (jhuleatt) [inline storage-extract-image-text/PREINSTALL.md:1]:
```suggestion
This extension will extract text from any `jpg` or `png` images uploaded to Cloud Storage and write the extracted text to Firestore.
```

File: storage-extract-image-text/PREINSTALL.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# Billing
+
+This extension uses other Firebase or Google Cloud Platform services which may have associated charges:
+
+<!-- List all products the extension interacts with -->
+
+- Cloud Functions
+- Cloud Vision API
+- Firebase Storage
+- Firebase Firestore
```
Comment (jhuleatt) [inline storage-extract-image-text/PREINSTALL.md:12]:
```suggestion
- Cloud Storage
- Cloud Firestore
```

**Review by jhuleatt — APPROVED** [review]

**Pull request merged**